android/app/src/main/java/io/peggyjo/peggymeter/HistoryTextFragment.java

```diff
@@ -31,12 +38,44 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
         dataController.addMoodListener(this);
         // Inflate the layout for this fragment
         View view = inflater.inflate(R.layout.fragment_history_text, container, false);
+        table = view.findViewById(R.id.LOG_ENTRIES_CONTAINER);
+        table.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
+            @Override
+            public boolean onItemLongClick(AdapterView<?> parent, View view, int position, long id) {
+                //used instead of getSelectedItem because getSelectedItem causes a null pointer exception
+                LogEntry selectedEntry = (LogEntry)table.getItemAtPosition(position);
+                EditText note = new EditText(getContext());
+
+                if (!selectedEntry.getComment().isEmpty()) {
+                    note.setText(selectedEntry.getComment());
+                }
+                AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+
+                if (!selectedEntry.getComment().isEmpty()) {
+                    builder.setTitle("Edit note");
+                } else {
+                    builder.setTitle("Add a note");
+                }
+                builder.setView(note);
+                AlertDialog editor = builder.create();
+                editor.setButton(-3, "Done", new DialogInterface.OnClickListener() {
```

Comment:
what's -3?

Reply:
A comment on "-3" wouldn't harm.

Reply:
I changed that parameter to AlertDialog.BUTTON_NEUTRAL, which was the proper AlertDialog reference with the same value.